src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/WorkloadSuggestionFinder.cs

```diff
@@ -0,0 +1,183 @@
+            {
+                foreach (Comparison<T> c in comparators)
+                {
+                    var cmp = c(val, best);
+                    if (cmp > 0)
+                    {
+                        best = val;
+                        break;
+                    }
+                    else if (cmp < 0)
+                    {
+                        break;
+                    }
+                }
+            }
+            return best;
+        }
+
+        /// <summary>
+        /// Gets the suggestion with the lowest number of extra packs and lowest number of workload IDs.
+        /// </summary>
+        public WorkloadSuggestion GetBestSuggestion() => FindBest(
+                UnsortedSuggestions,
+                (x, y) => y.ExtraPacks - x.ExtraPacks,
+                (x, y) => y.Workloads.Count - x.Workloads.Count);
+
+        /// <summary>
+        /// A partial or complete suggestion for workloads to install, annotated with which requested packs it does not satisfy
+        /// </summary>
+        class WorkloadSuggestionCandidate : IEquatable<WorkloadSuggestionCandidate>
+        {
+            public WorkloadSuggestionCandidate(HashSet<WorkloadDefinitionId> id, HashSet<WorkloadPackId> packs, HashSet<WorkloadPackId> unsatisfiedPacks)
+            {
+                Packs = packs;
+                UnsatisfiedPacks = unsatisfiedPacks;
+                Workloads = id;
+            }
+
+            public HashSet<WorkloadDefinitionId> Workloads { get; }
+            public HashSet<WorkloadPackId> Packs { get; }
+            public HashSet<WorkloadPackId> UnsatisfiedPacks { get; }
+            public bool IsComplete => UnsatisfiedPacks.Count == 0;
+
+            public bool Equals(WorkloadSuggestionCandidate? other) => other != null && Workloads.SetEquals(other.Workloads);
+
+            public override int GetHashCode()
+            {
+                int hashcode = 0;
+                foreach (var id in Workloads)
+                {
+                    hashcode ^= id.GetHashCode();
+                }
+                return hashcode;
+            }
+        }
+
+        /// <summary>
+        /// A suggestion of one or more workloads to be installed to satisfy missing packs.
+        /// </summary>
+        public class WorkloadSuggestion
+        {
+            public WorkloadSuggestion(HashSet<WorkloadDefinitionId> workloads, int extraPacks)
+            {
+                Workloads = workloads;
+                ExtraPacks = extraPacks;
+            }
+
+            /// <summary>
+            /// The workload IDs that comprise this suggestion
+            /// </summary>
+            public HashSet<WorkloadDefinitionId> Workloads { get; internal set; }
+
+            /// <summary>
+            /// How many additional (and potentionally unnecessary) packs this suggestion will result in installing
+            /// </summary>
+            public int ExtraPacks { get; internal set; }
```

Comment:
> How many additional (and potentionally unnecessary)

I don't understand what that means. The number could be too high due to some of the packs are unnecessary? And I need to read the comments to understand what the property does. It is a complex concept, it is fine to have a longer name.